Rpackage/R/mmit.R

```diff
@@ -0,0 +1,66 @@
+#' The Max Margin Interval Tree
+#'
+#' Learning a regression tree for censored data.
+#'
+#' @param target.mat The response variable of the model
+#' @param feature.mat a data frame containing the feature variables in the model.
+#' @param margin margin paramaters 
+#' @param loss The type of loss; (\code{"hinge"}, \code{"square"})
+#' @param maxdepth The maximum depth criteia
+#' @param min_sample The minimum number of sample required 
+#' 
+#' @return The learned regression tree as an object of class party.
+#' 
+#' @author Toby Dylan Hocking, Alexandre Drouin, Torsten Hothorn, Parismita Das
+#' 
+#' @examples
+#' library(mmit)
+#' target.mat <- rbind(
+#'   c(0,1), c(0,1), c(0,1),
+#'   c(2,3), c(2,3), c(2,3))
+#' 
+#' feature.mat <- rbind(
+#'   c(1,0,0), c(1,1,0), c(1,2,0),
+#'   c(1,3,0), c(1,4,0), c(1,5,0))
+#' 
+#' colnames(feature.mat) <- c("a", "b", "c")
+#' feature.mat <- data.frame(feature.mat)
+#' 
+#' 
+#' out <- mmit(target.mat, feature.mat)
+#' 
+#' @export
+mmit <- structure(function(target.mat, feature.mat, weights = NULL,
+                           depth=0, maxdepth = Inf, margin=0.0, loss="hinge",
+                           id = 1L, min_sample = 1) {
+  
+  ### arranging feature.mat.
+  feature.mat <- model.frame(data = feature.mat)
+  
+  ### assigning all weights as 1 in the beginning. weights determine which data is to be considered in which node.
+  if (is.null(weights)) weights <- rep(1L, nrow(feature.mat))
```

Comment:
Just initialize weights to a vector of ones. No need for more verifications. I don't see why weights should be user specified.

Reply:
yeah....will do that